examples/elasticsearch-incremental/elasticsearch-incremental-blueprint.yaml

```diff
@@ -0,0 +1,555 @@
+
+          while curl -k -u "elastic:$PASSWORD" -X GET "${ES_URL}/_snapshot/k10_repo/$SNAPSHOT_NAME/_status?pretty" | grep -P "(IN_PROGRESS|STARTED)"
+          do 
+            bp_echo "snapshot $SNAPSHOT_NAME still in progress"
+            size_in_bytes=$(curl -k -u "elastic:$PASSWORD" -X GET "${ES_URL}/_snapshot/k10_repo/$SNAPSHOT_NAME/_status?pretty" |grep size_in_bytes)
+            bp_echo $size_in_bytes
+            sleep 4
+          done 
+
+          if  curl -k -u "elastic:$PASSWORD" -X GET "${ES_URL}/_snapshot/k10_repo/$SNAPSHOT_NAME/_status?pretty" | grep SUCCESS
+          then 
+             bp_echo "snapshot $SNAPSHOT_NAME was successful"
+          else
+             bp_echo "snapshot $SNAPSHOT_NAME was not successful"
+             reason=$(curl -k -u "elastic:$PASSWORD" -X GET "${ES_URL}/_snapshot/k10_repo/$SNAPSHOT_NAME/_status?pretty")
+             bp_echo $reason
+             exit 1
+          fi
+  restore:
+    inputArtifactNames: 
+    - s3Snap  # use it with  for instance {{ .ArtifactsIn.s3Snap.KeyValue.repoPath }} or {{ .ArtifactsIn.s3Snap.KeyValue.snapshotName }}
+    phases:   
+    - func: Wait
+      name: waitElasticGreen
+      args:
+        timeout: 360s
+        conditions:
+          anyOf:
+            - condition: '{{ if  or (eq "{ $.status.health }" "green") (eq "{ $.status.health }" "yellow")}}true{{ else }}false{{ end }}'            
+              objectReference:
+                apiVersion: v1
+                group: elasticsearch.k8s.elastic.co
+                resource: elasticsearches
+                name: "{{ .Object.metadata.name }}"
+                namespace: "{{ .Object.metadata.namespace}}" 
+    - func: ScaleWorkload
+      name: ensureSTSIsScaled
+      args:
+        namespace: "{{ .Object.metadata.namespace}}"  
+        kind: statefulset
+        name: "{{ .Object.metadata.name }}-es-{{ (index .Object.spec.nodeSets 0).name }}"
+        replicas: "{{ (index .Object.spec.nodeSets 0).count }}"  
+    - func: KubeTask
+      name: setupNodeKeyStore
+      objects:
+        elasticSecret:
+          kind: Secret
+          name: '{{ .Object.metadata.name }}-es-elastic-user'
+          namespace: '{{ .Object.metadata.namespace }}' 
+      args:        
+        image: ghcr.io/kanisterio/kanister-kubectl-1.18:0.81.0
+        command:
+        - bash
+        - -x
+        - -o
+        - errexit
+        - -o
+        - pipefail
+        - -c
+        - |
+          debug=0
+          function bp_dbg
+          {
+              # use 
+              # kubectl logs -n kasten-io -l component=kanister --tail=10000 -f | ggrep -o -P '(?<=Elasticbps3echobegin).*(?=Elasticbps3echoend)' 
+              # to grab the logs in kanisters
+              if [[ $debug -eq 1 ]]
+              then 
+                echo "Elasticbps3echobegin backup-action: $* Elasticbps3echoend"
+              fi
+          }
```

Comment:
I think we should remove this as well, we are not using this anywhere in this phase.

Reply:
same goes with other phases where we have this function but are not using them.